server/blindai_sgx/src/client_communication.rs

```diff
@@ -255,36 +267,25 @@ impl Exchange for Exchanger {
 
         // Find the model with model_id
 
-        let uuid = match Uuid::from_str(&model_id) {
-            Ok(uuid) => uuid,
-            Err(_) => return Err(Status::invalid_argument("Model doesn't exist")),
-        };
+        let uuid = Uuid::from_str(&model_id).map_err(|_err| Status::invalid_argument("Model doesn't exist"))?;
 
         let res = self.model_store.use_model(uuid, |model| {
             (
-                model.run_inference(&mut input.clone()[..]),
-                model.model_name().map(|s| s.to_string()),
-                model.datum_output(),
+                model.run_inference(&input),
+                model.model_name().map(|e| e.to_string()),
+                model.datum_outputs().to_vec(),
```

Comment:
This PR also adds Multiple Outputs support. This is desirable, since we don't have this currently, but you should add an integration test for it.

Reply:
Also, i don't think the protobuf is updated for this support & the python client, or maybe i missed it